### ISSUE TITLE
The encode part of SAM fo onnx.

### ISSUE THREAD

**Issue by jay757425789** [ChaoningZhang/MobileSAM#87]
Hello, thanks for your significant work. I am employing it to embed devices for my application. However, I need help finding the encode part of SAM for onnx (the decode part is available). So, can you please offer it?
Best regards.

**Comment by ryouchinsa** [comment]
You can find the converter for the encoding part to the onnx format in this repository [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper).

Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)
